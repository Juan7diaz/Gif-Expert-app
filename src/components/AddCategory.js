import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories, setCantGifs }) => {

    //almacena lo estro por el usuario
    const [InputValue, setInputValue] = useState('')

    //obtiene lo el usuario escribiÃ³ en el input
    const handleInputChange = ( e ) =>{
        setInputValue( e.target.value );
    }
    
    //cuando el usuario presiona el 'Enter'
    const handleSubmit = ( e ) => {
        e.preventDefault(); //hace que el navegador no se recargue cuando hacemos un 'Enter'

        //para que no se garde texto mayor a 2 letras
        if( InputValue.trim().length > 2 ){
            setCategories( cats => [InputValue, ...cats] ); //se utiliza asi ya que no tengo la 'categories' (auqneu tambien la podemos pasar como props)
            setInputValue('');
        }
    }
    
    const handleCantChange = (e) => {
        setCantGifs( e.target.value )
    }

    return (
        <form onSubmit={ handleSubmit } className="form-inline">
            {/* <p>{ InputValue }</p> */}
            <input
                value = { InputValue }
                className="form-control mr-sm-2"
                onChange = { handleInputChange }
                placeholder="Ingrese el gif a buscar"
            />
            <select onChange={ handleCantChange }>
                <option value="5">5</option>
                <option value="10">10</option>
                <option value="20">20</option>
                <option value="30">30</option>
            </select>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
    setCantGifs: PropTypes.func.isRequired
}

export default AddCategory;