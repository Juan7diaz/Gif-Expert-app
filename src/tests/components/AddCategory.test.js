import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory'

describe('Testing en el componente addCategoy.js', () => {

    const setCategories = jest.fn(); //solo una funcion que no hace nada
    const setCantGifs = () => {} //solo es una funcion vacia x2
    let wrapper = shallow(  <AddCategory setCantGifs={setCantGifs} setCategories={setCategories}/> );
    
    beforeEach( ()=>{
        jest.clearAllMocks() //para que si tenemos alguna simulacion de funciones se limpie
        wrapper = shallow(  <AddCategory setCantGifs={setCantGifs} setCategories={setCategories}/> );
    })

    test('debe de mostrar el componente correctamente ', () => {
            
        expect( wrapper ).toMatchSnapshot();

    })
    
    test('debe de cambiar la caja de texto ', () => {
        
        const input = wrapper.find('input');
        //simulamos que la caja de texto cambió
        const value  = 'algun gifs';
        //lo que mande en {} va a tener el valor de 'e'  en const handleInputChange = ( e ) =>{...}
        input.simulate('change', { target: { value } } );
        //expect( wrapper.find('p').text().trim() ).toBe( value ) //en addCategoy.js comenté la linea del parrafo
    })
    
    test('No debe de post la informacion onSumit ', () => {
      
        wrapper.find('form').simulate('submit', { preventDefault(){} } )
        expect( setCategories ).not.toHaveBeenCalled();

    })
    
    test('debe de llamar setCategory y limpiar la caja de texto ', () => {
        
        //simpula el inputChange
        const value = 'HOLA MUNDO';
        wrapper.find('input').simulate('change', { target: { value } } );
        //expect( wrapper.find('p').text().trim() ).toBe( value )  //en addCategoy.js comenté la linea del parrafo
        
        //simula el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} } )
        
        //el setCategory fue llamado por lo menos una vez
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1); //que se halla llamado una vez
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) ); //que setCategories se halla llamado como una funcion

 
        //el valor del inputvalue debe ser ''
        const inputValue = wrapper.find('input').prop('value')
        expect( inputValue ).toBe('');

    })
    


})
