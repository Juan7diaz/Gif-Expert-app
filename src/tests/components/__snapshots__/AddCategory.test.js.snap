// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing en el componente addCategoy.js debe de mostrar el componente correctamente  1`] = `
<form
  className="form-inline"
  onSubmit={[Function]}
>
  <input
    className="form-control mr-sm-2"
    onChange={[Function]}
    placeholder="Ingrese el gif a buscar"
    value=""
  />
  <select
    onChange={[Function]}
  >
    <option
      value="5"
    >
      5
    </option>
    <option
      value="10"
    >
      10
    </option>
    <option
      value="20"
    >
      20
    </option>
    <option
      value="30"
    >
      30
    </option>
  </select>
</form>
`;
